### ISSUE TITLE
[dev.multiple-integrations] Proposal: integrations v2 should send auto-collected metrics to a specific instance

### ISSUE THREAD

**Issue by rfratto** [grafana/agent#1183]
Today, integrations work by dynamically creating a metrics instance for every single integration. Without instance sharing, which is scheduled to be deprecated (#398), this has a significant overhead when running many integrations, as each integration would receive its own dedicated WAL. 

Work has been ongoing in a dev.multiple-integrations branch which recently introduced a new iteration of the integrations subsystem. This new subsystem currently works by generating one single instance for all running integrations. While this is better, it still has three problems:

1. It still has _some_ overhead. This overhead can be avoided by disabling self-scraping and manually modifying an instance to use the new SD API to collect integrations. However, it's an annoying workaround that shouldn't have to be required to reduce resource usage. 

2. More features being added to the agent work by referencing an existing instance, such as auto-logging traces. This leaves the integration mechanism of creating a new instance feel weirdly out of place. 

3. It is unclear to users why `prometheus_remote_write` exists in integrations, which will be extra unclear once we start creating non-metrics integrations.

With integrations in the middle of a re-write which will be hidden behind a feature flag, I propose that we use the opportunity to align the rewrite with the rest of the codebase and have auto-collect metrics work by referencing a metrics instance defined in the config file.  

The schema would be as follows:

```yaml
integrations:
  # Config for metrics-based integrations.
  metrics:
    # Automatic scraping defaults. Individual instances of integrations 
    # may override these. 
    autoscrape: 
      [enable: <boolean> | default = true] 
      [metrics_instance: <string> | default = "default" ] 
      [scrape_interval: <duration> | default = <metrics.global.scrape_interval> ]  
      [scrape_timeout: <duration> | default = <metrics.global.scrape_timeout> ]  
    
  # ... integration configs follow ... 
```

The instance autoscraped metrics will be sent to defaults to "default." If a metrics instance with that name doesn't exist, the config will fail to load.

Additionally, the base set of settings for metrics-based integrations will change to the following schema:

```yaml
# Automatic scraping settings. Overrides defaults. Unlike the common autoscrape 
# settings, this includes relabeling rules.
autoscrape:
  [enable: <boolean> | default = <integrations.metrics.autoscrape.enable> ] 
  [metrics_instance: <string> | default = <integrations.metrics.autoscrape.metrics_instance> ] 
  [scrape_interval: <duration> | default = <integrations.metrics.autoscrape.scrape_interval> ]  
  [scrape_timeout: <duration> | default = <integrations.metrics.autoscrape.scrape_timeout> ]
  
  # Relabel and metric relabel configs used for auto-scraping.
  relabel_configs: 
    - [<relabel_config>]  
  metric_relabel_configs:
    - [<relabel_config>]

# Explicitly override the identifier for this integration instance. 
# If not provided, defaults to an integration-specific setting. 
[instance: <string>]
```

This allows people to send self-collected integrations to another remote_write endpoint, which is a neat capability they don't have today.

Internally, the integrations v2 subsystem will manually manage a single Prometheus SD component and a Prometheus Scraper which will write metrics to the appropriate preexisting WAL.

**Comment by rfratto** [comment]
One drawback to keep in mind: the shimmed v1 configs wouldn't have this new schema, meaning we'd want to move everything to v2 _before_ this stops becoming an experimental feature.

**Comment by rgeyer** [comment]
Having discussed this, I'm on-board with the changes.

The net result of the config changes is that they make more sense.

Given the feature flag we can stage this, implementing it where we need to (agent operator, and likely Grafana Cloud onboarding configs), and give enough lead-time for tooling and/or education of legacy customers.

Makes sense to me. 🚀

**Comment by rfratto** [comment]
Closed by #1195.
